skafka/src/main/scala/com/evolutiongaming/skafka/Converters.scala

```diff
@@ -95,20 +95,15 @@ object Converters {
   }
 
   implicit class MapJOps[K, V](val self: MapJ[K, V]) extends AnyVal {
-
     def asScalaMap[F[_]: Monad, A, B](ka: K => F[A], vb: V => F[B], keepNullValues: Boolean): F[Map[A, B]] = {
       self
         .asScala
         .toList
-        .collect {
-          case (k, v) if k != null && (keepNullValues || v != null) =>
-            for {
-              a <- ka(k)
-              b <- vb(v)
-            } yield (a, b)
+        .traverseFilter { case (k, v) => 
+          if (k != null && (keepNullValues || v != null)) (ka(k), vb(v)).mapN((_, _).some)
+          else none[(A, B)].pure[F]
```

Comment:
Due to Scala 3 bug, [compiler crashed here](https://github.com/lampepfl/dotty/issues/18769), so I rewrote without partial functions.

Reply:
please add comment about this bug.